src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1286,16 +1286,21 @@
 (defn update-latest-report!
   "Given a node name, updates the `certnames` table to ensure that it indicates the
    most recent report for the node."
-  [node]
-  {:pre [(string? node)]}
-  (let [latest-report (:id (first (query-to-vec
-                                    ["SELECT id FROM reports
+  [node report-id report-timestamp]
+  {:pre [(string? node) (integer? report-id)]}
+
+  (let [latest-report-timestamp (:latest_report_timestamp (first (query-to-vec
+                                    ["SELECT latest_report_timestamp
+                                      FROM certnames
                                       WHERE certname = ?
-                                      ORDER BY producer_timestamp DESC
-                                      LIMIT 1" node])))]
-    (jdbc/update! :certnames
-                  {:latest_report_id latest-report}
-                  ["certname = ?" node])))
+                                      " node])))]
+    (if (sql-ts-after? (to-timestamp report-timestamp)
+                     (to-timestamp latest-report-timestamp))
+
+      (jdbc/update! :certnames
+                    {:latest_report_id report-id
+                     :latest_report_timestamp report-timestamp}
+                    ["certname = ?" node]))))
```

Comment:
Can I avoid a round trip to the database above by adding 

```
["? > latest_report_timestamp" report-timestamp]
```

This would always send the UPDATE but the where clause would exclude cases where the new report timestamp isn't newer than the existing latest_report_timestamp.